src/client.rs

```diff
@@ -71,7 +72,7 @@ where
     let mut checkpoints = checkpoints
         .iter()
         .filter_map(
-            |checkpoint| match blockchain.storage.get_block(&checkpoint) {
```

Comment:
If the lock's `read()` failure here always means something fatal (another thread has panicked while holding the lock?), this looks OK. Otherwise, I'd rather invest some time in converting these errors into domain errors.